casemanagementv1/case_management_v1_integration_test.go

```diff
@@ -0,0 +1,380 @@
+
+			Expect(err).To(BeNil())
+			Expect(detailedResponse.StatusCode).To(Equal(200))
+			Expect(*result.Number).To(Equal(caseNumber))
+		})
+
+		It("Successfully got a case with field filtering", func() {
+			shouldSkipTest()
+
+			options.SetFields([]string{"number", "severity"})
+			result, detailedResponse, err := service.GetCase(options)
+
+			Expect(err).To(BeNil())
+			Expect(detailedResponse.StatusCode).To(Equal(200))
+			Expect(*result.Number).To(Equal(caseNumber))
+			Expect(result.Severity).To(Not(BeNil()))
+			Expect(result.Contact).To(BeNil())
+		})
+
+		It("Failed to get a case with bad params", func() {
+			shouldSkipTest()
+
+			options.SetFields([]string{"invalid_field"})
+			_, detailedResponse, err := service.GetCase(options)
+			Expect(err).To(Not(BeNil()))
+			Expect(detailedResponse.StatusCode).To(Not(Equal(200)))
+		})
+	})
+
+	Describe("Add comment", func() {
+		// var options *casemanagementv1.AddCommentOptions
+		options := service.NewAddCommentOptions(caseNumber, commentValue)
+
+		// BeforeEach(func() {
+		// })
+		It("Successfully added a comment to a case", func() {
+			shouldSkipTest()
+			result, detailedResponse, err := service.AddComment(options)
+
+			Expect(err).To(BeNil())
+			Expect(detailedResponse.StatusCode).To(Equal(200))
+			Expect(*result.Value).To(Equal(commentValue))
+			Expect(result.AddedAt).To(Not(BeNil()))
+			Expect(result.AddedBy).To(Not(BeNil()))
+		})
+	})
+
+	Describe("Add watchlist", func() {
+		options := service.NewAddWatchlistOptions(caseNumber)
+		options.SetWatchlist(watchlistPayload.Watchlist)
+
+		It("Successfully added users to case watchlist", func() {
+			shouldSkipTest()
+			result, detailedResponse, err := service.AddWatchlist(options)
+
+			Expect(err).To((BeNil()))
+			Expect(detailedResponse.StatusCode).To(Equal(200))
+			Expect(len(result.Added)).To(Equal(len(watchlistPayload.Watchlist)))
+		})
+	})
+
+	Describe("Remove watchlist", func() {
+		options := service.NewRemoveWatchlistOptions(caseNumber)
+		options.SetWatchlist(watchlistPayload.Watchlist)
+
+		It("Successfully removed users from case watchlist", func() {
+			shouldSkipTest()
+			_, detailedResponse, err := service.RemoveWatchlist(options)
+
+			if err != nil {
+				fmt.Printf("Remove watchlist response: %s\n", string(detailedResponse.RawResult))
+			}
+
+			Expect(err).To((BeNil()))
+			Expect(detailedResponse.StatusCode).To(Equal(200))
+		})
+	})
+
+	Describe("Update status", func() {
+		
+		It("Succefully resolve a case", func() {
+			shouldSkipTest()
+			resolvePayload, _ := service.NewResolvePayload(casemanagementv1.ResolvePayload_Action_Resolve, 1)
+			options := service.NewUpdateCaseStatusOptions(caseNumber, resolvePayload)
+
+			result, detailedResponse, err := service.UpdateCaseStatus(options)
+			
+			Expect(err).To(BeNil())
+			Expect(detailedResponse.StatusCode).To(Equal(200))
+			Expect(*result.Status).To(Equal("Resolved"))
+		})
+
+	It("Succefully unresolve a case", func() {
```

Comment:
Just a heads up. Support org recently submitted a request as below.

- Customers should not be able to set the resolution code. 
- Resolution codes should be hidden from customers.
- Only employees can set the resolution code.

We will have to make necessary changes to our API and to this test case.

Reply:
Thank you, Chong. Will modify the test case accordingly to how the API behaves in the future.